src/integral.lisp

```diff
@@ -115,15 +117,22 @@
 
 (defmethod make-insert-sql ((obj <dao-class>))
   (insert-into (intern (table-name obj) :keyword)
-    (apply #'set=
-           (mapcan
-            #'(lambda (slot-name)
+      (make-set-clause obj)))
+
+(defun make-set-clause (obj)
+  (apply #'set=
+         (mapcan
+          #'(lambda (slot)
+              (let ((slot-name (c2mop:slot-definition-name slot))
+                    (deflate-fn (table-column-deflate slot)))
                 (if (slot-boundp obj slot-name)
-                    (list (intern (symbol-name (unlispify slot-name)) :keyword)
-                          (deflate obj slot-name
-                            (slot-value obj slot-name)))
-                    nil))
-            (database-column-slot-names (class-of obj))))))
+                    (let ((value (slot-value obj slot-name)))
+                      (list (intern (symbol-name (unlispify slot-name)) :keyword)
+                            (if deflate-fn
+                                (funcall deflate-fn value)
+                                (deflate obj slot-name value))))
```

Comment:
It's not obvious that `deflate` method would be ignored if `:deflate` is declared in `defclass`.
It would be better to unify the logic into `deflate` method and to change the default behaviour to look if `:deflate` exists or not.